### ISSUE TITLE
Add rule for performant transitions

### ISSUE THREAD

**Issue by wuz** [stylelint/stylelint#3938]
Hello! I added this rule to CSSLint a while back and I would love to get it added in here as well. Here is a basic idea:

Using transition values that affect non-compositing rendering in the browser causes performance issues. For example, consider this div that moves 15px to the right on hover:

```css
div {
    transition: right 0.5s linear;
    position: absolute;
    right: 15px;
}
div:hover {
    right: 0px;
}
```

This transition requires the browser to do a re-layout, calculating the position of all elements. This transition would be much better suited to a transform, such as:

```css
div {
    transition: right 0.5s linear;
    position: absolute;
    right: 15px;
}
div:hover {
    transform: translateX(15px);
}
```

Using performant transitions makes animations run at 60fps and look smooth across browsers.

## Rule Details

Rule ID: `performant-transitions`

This rule is aimed at creating performant web animations. As such, it warns when `transition`, `-webkit-transition`, `-ms-transition`, and `transition-property` are used with values other than `transform` and `opacity`.

The following patterns are considered warnings:

```css
div {
    transition: width 0.5s linear;
}

div {
   transition: left 0.5s ease-in-out;
}
```
The following patterns are considered okay and do not cause warnings:

```css

/* using transform */
div {
    transition: transform 0.2s linear;
}

/* using opacity */
div {
   transition: opacity 0.5s ease-in-out;
}
```

**Comment by jeddy3** [comment]
@wuz Thank you for the suggestion and for the clear write-up!

I believe you can already enforce this practice with:

```json
{
  "rules": {
    "declaration-property-value-whitelist": {
      "/^transition/": ["/^opacity/", "/^transform/"]
    }
  }
}
```

From the [vision doc](https://github.com/stylelint/stylelint/blob/master/VISION.md#limit-language-features):

> Provide rules to limit what language features can be used, including:
> - Enforcing best practices _at the configuration level_ e.g. disallowing the all keyword for transitions as it is not performant.

We wanted to avoid basing rules around practices as they can become outdated and redundant. Instead, we created a [suite of rules that allow users to limit the use of CSS language features](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md#limit-language-features) at the configuration level. I believe this suite allows a user to target nearly every construct in the CSS language. You can do all sorts of cool things with these rules e.g. simplifying the CSS language itself by only whitelisting a subset it agreed by your team.

Alternatively, you can [create your own plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) and [share that with the community](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/plugins.md).

I wonder if adding this example to [the FAQ](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/faq.md) will help other users looking to achieve the same thing. What do you think?

**Comment by jeddy3** [comment]
> Alternatively, you can [create your own plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) and [share that with the community](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/plugins.md).

It turns out, there's already [a plugin for this](https://github.com/kristerkari/stylelint-high-performance-animation).

**Comment by wuz** [comment]
Thanks for the info! Makes sense to me 😄
